float32 width
float32 height
float32 size
float32 curve
float32 rotation
float32 offset_x
float32 offset_y